eng/common/scripts/modules/ChangeLog-Operations.psm1

```diff
@@ -0,0 +1,109 @@
+# Common Changelog Operations
+
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)\s(?<releaseStatus>\(Unreleased\)|\(\d{4}-\d{2}-\d{2}\))?)"
+
+# given a CHANGELOG.md file, extract the relevant info we need to decorate a release
+function Extract-ReleaseNotes {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$ChangeLogLocation,
+        [String]$VersionString
+    )
+    
+    $ErrorActionPreference = 'Stop'
+
+    $releaseNotes = @{}
+    if ($ChangeLogLocation.Length -eq 0)
+    {
+        return $releaseNotes
+    }
+    
+    try 
+    {
+        $contents = Get-Content $ChangeLogLocation
+    
+        # walk the document, finding where the version specifiers are and creating lists
+        $version = ""
+        foreach($line in $contents){
+            if ($line -match $RELEASE_TITLE_REGEX)
+            {
+               if ($version -ne "")
+               {
+                  $releaseNotes[$version].ReleaseContent = $releaseContentBuilder -join [Environment]::NewLine
```

Comment:
I would not join these as a newline just keep the array of strings. So I'd make ReleaseContent and ReleaseNoteEntry arrays.

Reply:
Ok, I can do that. I was again trying to keep things functioning in other places where this is being used.